chore: bump to 2.0.0

mdwcrft opened the pull request.

jerome-benoit commented:
It's way too early to release 2.0.0 given the PRs backlog: 

+ OCPP 2 enums namespace fix
+ OCPP 2 type definition 
+ ...

mdwcrft commented:
> It's way too early to release 2.0.0 given the PRs backlog: 
> 
> 
> 
> + OCPP 2 enums namespace fix
> 
> + OCPP 2 type definitions 
> 
> + Version support documentation
> 
> + ...  

Happy to make another release this week with those changes once merged. The aim of 2.0.0 is to officially release all changes since the last official release 0.26 which was 11 months ago

jerome-benoit commented:
> > It's way too early to release 2.0.0 given the PRs backlog:
> > 
> > * OCPP 2 enums namespace fix
> > * OCPP 2 type definitions
> > * Version support documentation
> > * ...
> 
> Happy to make another release this week with those changes once merged. The aim of 2.0.0 is to officially release all changes since the last official release 0.26 which was 11 months ago

The last official release is 1.0.0 and was not synced with GitHub release. 

A major is expected to include every single deprecation change *BEFORE*, not after.

ajmirsky reviewed (changes requested):
please include pending PRs before releasing new library. 

And I agree with @jerome-benoit , since most people install via pypi, the last official release is 1.0.0, even if it wasn't marked in github as a release.

if anything, this chore should be to create a github release to match what was released to pypi as 1.0.0

jainmohit2001 commented:
Will merge the pending PR's before releasing 2.0.0

ajmirsky commented:
> Will merge the pending PR's before releasing 2.0.0

thanks! greatly appreciated!

ajmirsky commented:
@mdwcrft  @proelke @jerome-benoit once #699 is merged, any other things needed before a release? happy to help!

jainmohit2001 commented:
I don't see any changes coming as of now. We'll do a pre-release first and test things out e2e using our internal systems. We can go for a major release afterwards.

jainmohit2001 commented:
@mdwcrft Feel free to create another rc release. I'll try to expedite the testing process from my end.

jainmohit2001 commented:
Looking good.
I was able to migrate from OCPP 0.26.0 to OCPP 2.0.0rc4.
The migration was pretty easy, the IDE instantly picked up missing deprecated classes and enums. Bulk renaming for call, call results and enums helped me complete the migration in a few minutes.
Tested a successful charging flow on both OCPP 1.6 and OCPP 2.0.1 protocol.
@mdwcrft we can go with a major release.
